### PR TITLE
Addressing post-merge issues and comments

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Addressing post-merge issues and comments

### Issues Resolved
Addressing post-merge issues and comments from `https://github.com/opensearch-project/spring-data-opensearch/pull/1`

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@rursprung please take a look, mostly addressing your comments, thank you

File: NOTICE.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ This product includes software developed at
 The Apache Software Foundation (http://www.apache.org/).
 
 This product includes/uses Spring Data Elasticsearch (https://github.com/spring-projects/spring-data-elasticsearch),
-developed by Spring team (https://github.com/spring-projects)
\ No newline at end of file
+developed by Spring team (https://github.com/spring-projects)
```
Comment (rursprung) [inline NOTICE.txt:7]:
```suggestion
developed by the Spring team (https://github.com/spring-projects).
```

File: README.md [hunk 2]
```diff
@@ -165,13 +165,26 @@ In this code snippet, the client configuration was customized to:
 - A `Supplier<Header>` function can be specified which is called every time before a request is sent to Opensearch - here, as an example, the current time is written in a header.
 - A function configuring the low level REST client
 
+### Spring Boot integration
+
+If you are using Spring Data Opensearch along with Spring Boot (3.x milestone releases), you may consider excluding the `ElasticsearchDataAutoConfiguration` configuration from automatic discovery.
```
Comment (rursprung) [inline README.md:170]: maybe say why "you may consider" doing that? and what happens if you don't do it?

File: README.md [hunk 3]
```diff
@@ -209,7 +222,7 @@ Attach a link to your code or a compressed file containing your code.
 
 You need [Apache Maven 3.5.0 or above](https://maven.apache.org/run-maven/index.html) and JDK 17 (or above) to build the `main` branch.
 
-```
+```bash
 $ mvn clean install
```
Comment (rursprung) [inline README.md:226]: if it's already tagged as `bash` it'll use bash-style syntax highlighting. having a lonely `$` somewhere in there would be wrong
```suggestion
mvn clean install
```

**Review by dblock — APPROVED** [review]
Rebase?

**Pull request merged**